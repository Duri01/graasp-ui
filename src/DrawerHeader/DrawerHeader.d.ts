import React from 'react';

interface DrawerHeaderProps {
  classes: {
    secondaryAction: string;
    drawerHeader: string;
  };
  theme: {
    direction?: string;
  };
  handleDrawerClose: () => {};
}

declare const DrawerHeader: React.FC<DrawerHeaderProps>;
