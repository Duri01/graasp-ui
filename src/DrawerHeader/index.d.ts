export { default } from './DrawerHeader';
export * from './DrawerHeader';
