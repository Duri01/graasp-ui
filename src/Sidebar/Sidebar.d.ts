import React from 'react'

interface SidebarProps {
  className?: string
  classes: {
    drawer: string
    drawerPaper: string
  }
  isSidebarOpen?: boolean
  handleDrawerClose: () => {}
  children?: React.ReactElement
  drawerHeaderContent?: React.ReactElement
}

declare const Sidebar: React.FC<SidebarProps>
