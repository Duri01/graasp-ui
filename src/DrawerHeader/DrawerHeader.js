import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import { DEFAULT_DIRECTION, DRAWER_HEADER_HEIGHT } from '../constants';

const styles = (theme) => ({
  drawerHeader: {
    height: DRAWER_HEADER_HEIGHT,
  },
  secondaryAction: {
    right: theme.spacing(1),
  },
});

export const DrawerHeader = ({
  classes,
  handleDrawerClose,
  children,
  theme: { direction = DEFAULT_DIRECTION },
}) => {
  return (
    <ListItem
      classes={{ root: classes.drawerHeader }}
      divider
      ContainerComponent='div'
    >
      {children}
      <ListItemSecondaryAction classes={{ root: classes.secondaryAction }}>
        <IconButton onClick={handleDrawerClose}>
          {direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  );
};

const StyledComponent = withStyles(styles, { withTheme: true })(DrawerHeader);

export default StyledComponent;
