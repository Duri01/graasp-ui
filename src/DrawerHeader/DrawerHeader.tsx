import React, { FC } from 'react';
import { Theme, createStyles, makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import { DEFAULT_DIRECTION, DRAWER_HEADER_HEIGHT } from '../constants';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    drawerHeader: {
      height: DRAWER_HEADER_HEIGHT,
    },
    secondaryAction: {
      right: theme.spacing(1),
    },
  }),
);

export interface DrawerHeaderProps {
  theme?: {
    direction?: string;
  };
  children?: JSX.Element;
  handleDrawerClose?: () => void;
}

export const DrawerHeader: FC<DrawerHeaderProps> = ({
  handleDrawerClose,
  children,
  theme,
}) => {
  const classes = useStyles();
  const direction = theme?.direction ?? DEFAULT_DIRECTION;
  return (
    <ListItem
      classes={{ root: classes.drawerHeader }}
      divider
      ContainerComponent='div'
    >
      {children}
      <ListItemSecondaryAction classes={{ root: classes.secondaryAction }}>
        <IconButton onClick={handleDrawerClose}>
          {direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  );
};

export default DrawerHeader;
