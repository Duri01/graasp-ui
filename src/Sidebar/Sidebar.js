import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Drawer from '@material-ui/core/Drawer'
import DrawerHeader from '../DrawerHeader'
import { DRAWER_WIDTH } from '../constants'

const styles = () => ({
  drawer: {
    width: DRAWER_WIDTH,
    flexShrink: 0
  },
  drawerPaper: {
    width: DRAWER_WIDTH
  }
})

export const Sidebar = ({
  className,
  classes,
  isSidebarOpen = false,
  handleDrawerClose,
  children
}) => {
  return (
    <Drawer
      className={className}
      variant='persistent'
      anchor='left'
      open={isSidebarOpen}
      classes={{
        paper: classes.drawerPaper
      }}
    >
      <DrawerHeader handleDrawerClose={handleDrawerClose} />
      {children}
    </Drawer>
  )
}

const StyledComponent = withStyles(styles, { withTheme: true })(Sidebar)

export default StyledComponent
