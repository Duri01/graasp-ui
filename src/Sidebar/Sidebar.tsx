import React, { FC } from 'react';
import Drawer from '@mui/material/Drawer';
import DrawerHeader from '../DrawerHeader';
import { DRAWER_WIDTH } from '../constants';
import { styled, SxProps } from '@mui/material';

export interface SidebarProps {
  sx?: SxProps;
  isSidebarOpen?: boolean;
  handleDrawerClose?: () => void;
  children?: React.ReactElement;
  drawerHeaderContent?: React.ReactElement;
}

export const Sidebar: FC<SidebarProps> = ({
  sx,
  isSidebarOpen = false,
  handleDrawerClose,
  children,
  drawerHeaderContent,
}) => {
  const SideBar = styled(Drawer)({
    width: DRAWER_WIDTH,
    flexShrink: 0,
    '.MuiDrawer-paper': {
      width: DRAWER_WIDTH,
    },
  });
  return (
    <SideBar sx={sx} variant='persistent' anchor='left' open={isSidebarOpen}>
      <DrawerHeader handleDrawerClose={handleDrawerClose}>
        {drawerHeaderContent}
      </DrawerHeader>
      {children}
    </SideBar>
  );
};

export default Sidebar;
