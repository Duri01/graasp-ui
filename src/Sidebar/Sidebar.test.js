import React from 'react';
import { shallow } from 'enzyme';
import Drawer from '@material-ui/core/Drawer';
import { Sidebar } from './Sidebar';
import DrawerHeader from '../DrawerHeader';

const createSidebarProps = ({
  className,
  isSidebarOpen,
  handleDrawerClose,
} = {}) => {
  return {
    className,
    classes: {
      drawer: '',
      drawerPaper: '',
    },
    isSidebarOpen,
    handleDrawerClose,
  };
};

describe('Sidebar', () => {
  let wrapper;

  describe('default values', () => {
    beforeAll(() => {
      const props = createSidebarProps();
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<Sidebar {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <Drawer /> and one <DrawerHeader /> components with default values', () => {
      const drawer = wrapper.find(Drawer);
      expect(drawer).toHaveLength(1);
      expect(drawer.prop('open')).toEqual(false);
      const drawerHeader = drawer.find(DrawerHeader);
      expect(drawerHeader).toHaveLength(1);
    });
  });

  describe('isSidebarOpen = true, className, handleDrawerClose, children are defined', () => {
    const mockFunction = jest.fn();
    const children = <child>some child</child>;

    beforeAll(() => {
      const props = createSidebarProps({
        isSidebarOpen: true,
        className: 'className',
        handleDrawerClose: mockFunction,
      });
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<Sidebar {...props}>{children}</Sidebar>);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <Drawer /> and one <DrawerHeader /> components with defined values', () => {
      const drawer = wrapper.find(Drawer);
      expect(drawer).toHaveLength(1);
      expect(drawer.prop('open')).toEqual(true);
      expect(drawer.prop('className')).toEqual('className');
      const drawerHeader = drawer.find(DrawerHeader);
      expect(drawerHeader).toHaveLength(1);
      expect(drawerHeader.prop('handleDrawerClose')).toBe(mockFunction);
      const childrenComponent = drawer.find('child');
      expect(childrenComponent).toHaveLength(1);
      expect(childrenComponent.text()).toContain('some child');
    });
  });
});
