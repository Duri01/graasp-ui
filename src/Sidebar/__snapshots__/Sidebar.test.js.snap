// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar default values renders correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  classes={
    Object {
      "paper": "",
    }
  }
  open={false}
  variant="persistent"
>
  <WithStyles(DrawerHeader) />
</WithStyles(ForwardRef(Drawer))>
`;

exports[`Sidebar isSidebarOpen = true, className, handleDrawerClose, children are defined renders correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="className"
  classes={
    Object {
      "paper": "",
    }
  }
  open={true}
  variant="persistent"
>
  <WithStyles(DrawerHeader)
    handleDrawerClose={[MockFunction]}
  />
  <child>
    some child
  </child>
</WithStyles(ForwardRef(Drawer))>
`;
